client/client-multi/client-c/src/kaa/kaa_notification_manager.h

```diff
@@ -0,0 +1,185 @@
+*
+* @return Error code.
+*/
+kaa_error_t kaa_add_optional_notification_listener(kaa_notification_manager_t *self, kaa_notification_listener_t *listener
+                                                 , uint64_t *topic_id, uint32_t *listener_id);
+
+/**
+* @brief Removes a mandatory notification listener
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  listener_id  Pointer to the listener id, which will be used to find listener that should be removed from notification listeners' list. If NULL listener_id won't be initialized
+*
+* @return Error code.
+*/
+kaa_error_t kaa_remove_notification_listener(kaa_notification_manager_t *self, uint32_t *listener_id);
+
+/**
+* @brief Removes a optional notification listener.
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  topic_id     Pointer to the id of topic listener don't want to be notified about.
+* @param[in]  listener_id  Pointer to the variable, which will be used to find listener that should be removed from notification listeners' list.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_remove_optional_notification_listener(kaa_notification_manager_t *self, uint64_t *topic_id, uint32_t *listener_id);
+
+/**
+* @brief Adds a topic list listener.
+*
+* @param[in]  self               Pointer to the notification manager instance.
+* @param[in]  listener           Pointer to a listener, which callback will be called as soon as a notification received.
+* @param[out] topic_listener_id  Pointer to a variable, which will be initialized with calculated id.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_add_topic_list_listener(kaa_notification_manager_t *self, kaa_topic_listener_t *listener, uint32_t *topic_listener_id);
+
+/**
+* @brief Removes the topic list listener.
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  topic_listener_id  Pointer to the integer variable, which will be used to find listener that should be removed from topic listeners' list.. If NULL topic_listener_id won't be initialized
+*
+* @return Error code.
+*/
+kaa_error_t kaa_remove_topic_list_listener(kaa_notification_manager_t *self, uint32_t *topic_listener_id);
+
+/**
+* @brief Retrieves topic list.
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  topics       Pointer to the pointer that will be initialized with topic list.
+*
+* @return Error code.
+*/
+kaa_error_t kaa_get_topics(kaa_notification_manager_t *self, kaa_list_t **topics);
+
+/**
+* @brief Subscribes to topic.
+*
+* @param[in]  self         Pointer to the notification manager instance.
+* @param[in]  topic_id     Pointer to topic id it is should be subscribed to.
+* @param[in]  force_sync   Indicates whether subscription should be done immediately (true) or could be postponed (false).
```

Comment:
Indicates whether subscription should be performed immediately (true) or may be postponed (false).